src/app-layout/classic.tsx

```diff
@@ -295,6 +277,24 @@ const ClassicAppLayout = React.forwardRef(
       placement.inlineSize - effectiveNavigationWidth - minContentWidth - contentPadding - rightDrawerBarWidth
     );
 
+    const getEffectiveToolsWidth = () => {
+      if (activeDrawerSize) {
+        return Math.min(resizableSpaceAvailable, activeDrawerSize);
```

Comment:
This is the relevant change, replacing `activeDrawerSize` with `Math.min(resizableSpaceAvailable, activeDrawerSize)`.

The `getEffectiveToolsWidth` function has been moved down in order to have access to `resizableSpaceAvailable`